src/components/spotlight/spotlight.story.js

```diff
@@ -0,0 +1,217 @@
+      table: {
+        category: 'Optional',
+      },
+    },
+    show_credit: {
+      name: 'credit',
+      type: { name: 'boolean' },
+      description: 'Show the credit',
+      table: {
+        category: 'Optional',
+      },
+    },
+    font_size: {
+      name: 'font size',
+      type: 'select',
+      description: 'Change font size',
+      options: ['m', 'l'],
+      control: {
+        labels: {
+          m: 'medium',
+          l: 'large',
+        },
+      },
+      mapping: {
+        medium: 'm',
+        large: 'l',
+      },
+      table: {
+        type: 'string',
+        defaultValue: { summary: 'm' },
+        category: 'Display',
+      },
+    },
+    full_width: {
+      name: 'full width',
+      type: 'boolean',
+      description: 'Take the full width of the viewport when in a container',
+      mapping: {
+        0: false,
+        1: true,
+      },
+      table: {
+        type: { summary: 'boolean' },
+        defaultValue: { summary: 'false' },
+        category: 'Display',
+      },
+    },
+    header: {
+      type: 'string',
+      description: 'Header of the spotlight',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
+        category: 'Content',
+      },
+      if: { arg: 'show_header' },
+    },
+    title: {
+      type: 'string',
+      description: 'Title of the spotlight',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
+        category: 'Content',
+      },
+    },
+    credit: {
+      type: 'string',
+      description: 'Credit of the image',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
+        category: 'Content',
+      },
+      if: { arg: 'show_credit' },
+    },
+    sidebarContent: {
+      name: 'demo sidebar content',
+      type: { name: 'boolean' },
+      description:
+        'Inject a test content block in the sidebar, to see the display',
+      table: {
+        category: 'Test content',
+      },
+      control: {
+        type: 'boolean',
+      },
+    },
+    gridContent: {
+      name: 'demo grid content',
+      type: { name: 'boolean' },
+      description:
+        'Inject a test content block displayed on the grid, to see the alignment',
+      table: {
+        category: 'Test content',
+      },
+      control: {
+        type: 'boolean',
+      },
+    },
+  };
+
+  if (data.picture) {
+    argTypes.image = {
+      type: 'string',
+      description: 'Path or Url of the background image',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
+        category: 'Content',
+      },
+    };
+  }
+
+  return argTypes;
+};
+
+const prepareData = (data, args) => {
+  correctPaths(data);
+  const clone = JSON.parse(JSON.stringify(data));
+  Object.assign(clone, args);
+
+  if (!args.show_header) delete clone.header;
+  if (!args.show_anchor) clone.has_anchor = 0;
```

Comment:
ahh, nice, this way you bypass the problem of the template using default(true), once you'll use the ?? you will be able to set this to false ;)